### PR TITLE
Added new method to ByteBuffer & Slice

### PR DESCRIPTION
**Pull request by veblush**
Implementing https://github.com/grpc/proposal/blob/master/L79-cpp-byte-buffer-slice-methods.md

### CODE REVIEW

File: include/grpcpp/impl/codegen/byte_buffer.h [hunk 1]
```diff
@@ -114,6 +114,13 @@ class ByteBuffer final {
     return *this;
   }
 
+  // If this cord's representation is a single flat slice, returns a
+  // slice referencing that array.
+  Status TrySingleSlice(Slice* slice) const;
+
+  /// Dump (read) the buffer contents into \a slices.
```
Comment (vjpai) [inline include/grpcpp/impl/codegen/byte_buffer.h:121]:
```suggestion
  /// Dump (read) the buffer contents into \a slice.
```

File: include/grpcpp/impl/codegen/byte_buffer.h [hunk 2]
```diff
@@ -114,6 +114,13 @@ class ByteBuffer final {
     return *this;
   }
 
+  // If this cord's representation is a single flat slice, returns a
```
Comment (vjpai) [inline include/grpcpp/impl/codegen/byte_buffer.h:117]: What's a `cord`?

File: test/cpp/util/byte_buffer_test.cc [hunk 3]
```diff
@@ -123,6 +123,35 @@ TEST_F(ByteBufferTest, SerializationMakesCopy) {
...
+  std::vector<Slice> slices;
+  slices.emplace_back(kContent1);
+  ByteBuffer buffer(&slices[0], 1);
+  Slice slice;
+  EXPECT_TRUE(buffer.TrySingleSlice(&slice).ok());
+  EXPECT_EQ(slice.size(), slices[0].size());
+  EXPECT_EQ(memcmp(slice.begin(), slices[0].begin(), slice.size()), 0);
+}
+
+TEST_F(ByteBufferTest, TrySingleSliceWithMultiplelices) {
```
Comment (vjpai) [inline test/cpp/util/byte_buffer_test.cc:136]:
```suggestion
TEST_F(ByteBufferTest, TrySingleSliceWithMultipleSlices) {
```

**Review by vjpai — APPROVED** [review]
LGTM with minor nits in the comments.

File: include/grpcpp/impl/codegen/byte_buffer.h (hunk 1 above)
Comment (veblush) [reply include/grpcpp/impl/codegen/byte_buffer.h]: Done

File: include/grpcpp/impl/codegen/byte_buffer.h (hunk 2 above)
Comment (veblush) [reply include/grpcpp/impl/codegen/byte_buffer.h]: Good catch. Done!

File: test/cpp/util/byte_buffer_test.cc (hunk 3 above)
Comment (veblush) [reply test/cpp/util/byte_buffer_test.cc]: Done

**Pull request merged**